pkg/controllers/nodeclaim/expiration/controller.go

```diff
@@ -87,6 +93,10 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (re
 	return reconcile.Result{}, nil
 }
 
+func (c *Controller) Name() string {
+	return "nodeclaim.expiration"
+}
+
```

Comment:
I've seen this used and not used, do we want to open a `good-first-issue` asking to implement this across all controllers?

Reply:
Sure, I think @jmdeal added this in a few places but didn't add it everywhere -- really, there's probably a way to wrap our Reconciles so that names are auto-injected without us having to explicitly do this anywhere

Reply:
Done: https://github.com/kubernetes-sigs/karpenter/issues/2001